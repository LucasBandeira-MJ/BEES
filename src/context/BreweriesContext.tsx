import { createContext, useState } from "react";


interface BreweriesContextData {
    user: string
    updateUser: (username: string) => void
}

interface BreweriesProviderProps {
    children: React.ReactNode
}

export const BreweriesContext = createContext({} as BreweriesContextData)

export const BreweriesProvider = ({children}: BreweriesProviderProps) => {
    const [user, setUser] = useState<string>('')
    
    const updateUser = (username: string) => {
        setUser(username)
    }

    return (
        <BreweriesContext.Provider value={{
            user,
            updateUser
        }}>
            { children }
        </BreweriesContext.Provider>
    )
}