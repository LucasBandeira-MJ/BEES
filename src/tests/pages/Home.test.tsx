import { render } from "@testing-library/react"
import { mocked } from 'jest-mock'
import { useRouter } from 'next/router'

import renderer from 'react-test-renderer'

import { BreweriesContext } from "../../context/BreweriesContext"
import Home from '../../pages'

jest.mock('next/router')

describe("Home component", () => {

    it('should match the snapshot', () => {
        const snapshot = renderer
            .create(<Home />)
            .toJSON()

            expect(snapshot).toMatchSnapshot()
    })

    it("should redirect user if user is set", () => {
        
        const useRouterMocked = mocked(useRouter) 
        const pushMocked = jest.fn()

        useRouterMocked.mockReturnValueOnce({
            push: pushMocked
        } as any)
        
        render(
            <BreweriesContext.Provider value={{ user: 'John Joe' } as any} >
                <Home />
            </BreweriesContext.Provider>
        )    

        expect(pushMocked).toHaveBeenCalled()        
    })
})