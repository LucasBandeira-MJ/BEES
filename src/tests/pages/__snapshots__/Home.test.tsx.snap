// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should match the snapshot 1`] = `
<div
  className="sc-gsnTZi hMJYIl"
>
  <form
    className="sc-bczRLJ ktGKjt"
    onSubmit={[Function]}
  >
    <p>
      Please enter your full name below
    </p>
    <p>
      Only alphabetical characters are accepted
    </p>
    <input
      onChange={[Function]}
      placeholder="Full name"
      required={true}
      type="text"
      value=""
    />
    <div
      className="field-control"
    >
      <input
        id="ageInput"
        onChange={[Function]}
        required={true}
        type="checkbox"
      />
      <label
        htmlFor="ageInput"
      >
        Are you older than 18 year old?
      </label>
    </div>
    <button
      disabled={true}
      type="submit"
    >
      Enter
    </button>
  </form>
</div>
`;
